lib/spack/spack/build_systems/oneapi.py

```diff
@@ -22,46 +24,44 @@ class IntelOneApiPackage(Package):
 
     phases = ['install']
 
-    def component_info(self,
-                       dir_name,
-                       components,
-                       releases,
-                       url_name):
+    def component_info(self, dir_name):
```

Comment:
Currently it looks like the only purpose of `dir_name` is to confirm that the component was installed successfully. If that's true, could you update this to be a property

e.g.

```
@property
def component_dir(self):
  raise NotImplementedError
```

the inheritors of the `IntelOneapiPackage` can define this property, and the install method can check

```
if not isdir(join(prefix, self.component_dir)):
```

that approach matches some common patterns in Spack packages when a build system requires a specific detail from a subclass.  Another option is to add a docstring here clarifying the use of `component_info` like

```
def component_info():
  """Define the expected installation directory. After the install script runs,
     IntelOneApiPackage will look for this directory to make sure the install
     was successful.
  """
```

Reply:
I like the first solution, but my understanding of python is that a base class cannot reference methods in the derived class. I will add the docstring.

Reply:
(no change needed) Regarding:

> I like the first solution, but my understanding of python is that a base class cannot reference methods in the derived class.

You can do this if desired in Python (the way I was suggesting doing it was somewhat analogous to defining an abstract class in Java, if that reference point is worth anything). Like I said though it's fine to just add a comment: that clarifies what the function is for.

Reply:
You are right. Using the component_dir solution. It improves clarity for the packages.